Library/Format.swift

```diff
@@ -384,7 +408,7 @@ private struct NumberFormatterConfig {
   fileprivate let currencySymbol: String
 
   fileprivate func formatter() -> NumberFormatter {
-    let formatter = NumberFormatter()
+    let formatter = AttributedNumberFormatter()
```

Comment:
My thinking behind this was that I didn't want to introduce a new formatter (cached formatter) since it would be identical to the `defaultCurrencyFormatter` which is already heavily covered by tests (these tests would be duplicate while there's no change to the functionality of the formatter other than extending it with attributed string support).

In order to set attributes call site can simply test for class type and cast this object properly.

I've done many iterations on this and at the end this seemed like a good tradeoff between keeping the functionality same (not duplicating tests) and adding attributed output support.

More than happy to discuss this